### PR TITLE
Added enterFullscreen & exitFullscreen methods for Windows & Linux

### PR DESCRIPTION
**Pull request by alexmercerind**
Added `enterFullscreen` & `exitFullscreen` to enter and exit seamless fullscreen mode (without titlebar).

**Usage**
- `enterFullscreen`
```dart
enterFullscreen();
```
- `exitFullscreen`
~Requires final window frame as `Rect`.~
```dart
exitFullscreen();
```

**Demo**

https://user-images.githubusercontent.com/28951144/121733126-b6b28480-cb10-11eb-9708-b5ac73d3a7ff.mp4

### CODE REVIEW

**Comment by alexmercerind** [comment]
@googlebot I signed it!

**Comment by alexmercerind** [comment]
I can try to add Linux support aswell.

**Comment by alexmercerind** [comment]
Added Linux support aswell. Looking forward for any requested changes to get this merged.

Thankyou.

File: plugins/window_size/windows/window_size_plugin.cpp [hunk 1]
```diff
@@ -245,6 +247,46 @@ void WindowSizePlugin::HandleMethodCall(
...
+    fullscreenSettings.dmBitsPerPel = GetDeviceCaps(hdc, BITSPIXEL);
+    fullscreenSettings.dmDisplayFrequency = GetDeviceCaps(hdc, VREFRESH);
+    fullscreenSettings.dmFields = DM_PELSWIDTH | DM_PELSHEIGHT | DM_BITSPERPEL | DM_DISPLAYFREQUENCY;
+    SetWindowLongPtr(GetRootWindow(registrar_->GetView()), GWL_EXSTYLE, WS_EX_APPWINDOW | WS_EX_TOPMOST);
+    SetWindowLongPtr(GetRootWindow(registrar_->GetView()), GWL_STYLE, WS_POPUP | WS_VISIBLE);
+    SetWindowPos(GetRootWindow(registrar_->GetView()), HWND_TOPMOST, 0, 0, GetDeviceCaps(hdc, DESKTOPHORZRES), GetDeviceCaps(hdc, DESKTOPVERTRES), SWP_SHOWWINDOW);
+    isChangeSuccessful = ChangeDisplaySettings(&fullscreenSettings, CDS_FULLSCREEN) == DISP_CHANGE_SUCCESSFUL;
+    ShowWindow(GetRootWindow(registrar_->GetView()), SW_MAXIMIZE);
+    result->Success(
+      flutter::EncodableValue(isChangeSuccessful)
```
Comment (stuartmorgan) [inline plugins/window_size/windows/window_size_plugin.cpp:267]: Why is failure being returned as a success? And why do the return semantics of Windows and Linux not match (nor does this match the use on the Dart side, which ignores the return value).

File: plugins/window_size/windows/window_size_plugin.cpp [hunk 2]
```diff
@@ -245,6 +247,46 @@ void WindowSizePlugin::HandleMethodCall(
...
+    bool isChangeSuccessful;
+    HWND hwnd = GetRootWindow(registrar_->GetView());
+    SetWindowLongPtr(hwnd, GWL_EXSTYLE, WS_EX_LEFT);
+    SetWindowLongPtr(hwnd, GWL_STYLE, WS_OVERLAPPEDWINDOW | WS_VISIBLE);
+    isChangeSuccessful = ChangeDisplaySettings(NULL, CDS_RESET) == DISP_CHANGE_SUCCESSFUL;
+    SetWindowPos(hwnd, HWND_NOTOPMOST, x, y, width, height, SWP_SHOWWINDOW);
+    ShowWindow(hwnd, SW_RESTORE);
+    result->Success(
+      flutter::EncodableValue(isChangeSuccessful)
+    );
```
Comment (stuartmorgan) [inline plugins/window_size/windows/window_size_plugin.cpp:289]: Please follow the Google style guide, and the style of the existing code. This code block is violating Google style rules for variable naming, and not following existing conventions of this code (not naming variables their types, specifying global namespace resolution for Win32 calls)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for the submission. The macOS implementation should be straightforward, so I can add that once this is finished.

**Comment by alexmercerind** [comment]
Now I've followed Google code style precisely.

Now `exitFullscreen` no longer requires `frame` parameter and window restores to initial size now after exiting fullscreen.

On windows, there isn't a specific API to use for fullscreen like GTK, so I had to do that.
Now I made it store initial window frame as `RECT` globally before going fullscreen and restore window to that frame upon exiting from fullscreen mode.
As a result, I also improved the Windows implementation by using [this](https://devblogs.microsoft.com/oldnewthing/20100412-00/?p=14353) example.

**Comment by alexmercerind** [comment]
@stuartmorgan what do i do now? 😟

**Comment by stuartmorgan** [comment]
@alexmercerind What do you do about what? I'm not sure what the question is.

**Comment by bdlukaa** [comment]
Any chance of this getting merged soon?

**Comment by stuartmorgan** [comment]
I need to have time to review it before it can be merged. If you can't wait until that has happened, you can use the PR's branch.

**Comment by sabin26** [comment]
@alexmercerind 
When windows button is pressed from keyboard to access the Taskbar after entering full screen, I am unable to view other applications after opening them from the taskbar. The flutter window stays on top of other windows. So, only flutter window is on the front view. Please have a look at it.

**Comment by stuartmorgan** [comment]
Just wanted to update to say that I haven't forgotten about this, I'm just balancing a lot of priorities, and context switching to two platforms I'm no longer regularly using requires a significant time investment.

/cc @robert-ancell and @clarkezone in case you have time to review your respective platforms of expertise here.

**Comment by stuartmorgan** [comment]
I apologize that this PR has been in limbo for so long.

After recently taking stock of all of the work across desktop and plugins in general, it's clear that expanding the scope of `window_size` with additional functionality isn't the best use of the Flutter team's resources, so I've [updated the plugin's README](https://github.com/google/flutter-desktop-embedding/blob/master/plugins/window_size/README.md#scope) to lay out the new, more restricted plan for the scope of the plugin. (The long delay on this PR demonstrates the problem with leaving an open-ended scope for a plugin that is in practice very low priority relative to all of the other work.)

Given that new scope, I'm going to close this PR. I encourage you to create a new plugin for this purpose, or to submit it to an existing community-developed window manipulation plugin.

(I understand that it may seem strange not to take an already-written PR based on resourcing considerations, but there is still non-trivial cost in taking the PR; in addition to review, there would undoubtedly be follow-up issues due to interaction with other parts of the API, edge cases with OS behaviors, etc., since full-screen is inherently complex. Having that in a plugin that will get the resources needed to maintain it more actively than we can currently maintain `window_size` will be a better experience for everyone.)

**Pull request closed without merging**